docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,292 @@
+definitions are _not_ part of the defining module's ABI.
+
+Most notably, default argument expressions are implicitly
+`@_alwaysEmitIntoClient`, which means that adding
+a default argument to a function which did not have one previously
+does not break ABI.
+
+## `@_alignment`
+
+## `@_borrowed`
+
+## `@_cdecl`
+
+Similar to `@_silgen_name` but uses the C calling convention.
+This attribute doesn't have very well-defined semantics.
+
+Type bridging is not done, so the parameter and return types should
+correspond directly to types accessible in C.
+In most cases, it is preferable to define a static method on an `@objc`
+class instead of using `@_cdecl`.
+
+For potential ideas on stabilization, see
+[Formalizing `@cdecl`](https://forums.swift.org/t/formalizing-cdecl/40677).
+
+## `@_distributedActorIndependent`
+
+## `@_disfavoredOverload`
+
+Marks a declaration that the type checker should try to avoid using. When the
+expression type checker is considering overloads, it will prefer a solution
+with fewer `@_disfavoredOverload` declarations over one with more of them.
+
+Historically, it was designed to help with `ExpressibleByXYZLiteral` types.
+The type checker strongly prefers to give literals their default type
+(e.g. `Int` for `ExpressibleByIntegerLiteral`,
+`String` for `ExpressibleByStringLiteral`, etc.).
+So if you want your API to prefer some other type but accept the default one
+too, you may need to mark the declaration taking the default literal type with
+this attribute:
+
+    extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
+
+    extension Text {
+        // We want `Text("foo")` to use this initializer:
+        init(_ key: LocalizedStringKey) { ... }
+
+        // But without @_disfavoredOverload, it would use this one instead,
+        // because that lets it give the literal its default type:
+        @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
+    }
+
+`@_disfavoredOverload` is a stop-gap design; eventually the type-checker
+behavior should be fixed and the attribute should be removed/made into a no-op.
+
+It's main use case is to work around clearly undesirable overload resolution,
+not to arbitrarily manipulate overload priority.
+
+## `@_dynamicReplacement(for:)`
+
+Marks a function as the dynamic replacement for another `dynamic` function.
+This is similar to method swizzling in other languages such as Objective-C,
+except that the replacement happens at program start (or loading a shared
+library), instead of at an arbitrary point in time.
+
+For more details, see [Dynamic method replacement](https://forums.swift.org/t/dynamic-method-replacement/16619).
+
+## `@_effects`
+
+## `@_exported`
+
+Use to mark an imported module to re-export all its declarations.
+This attribute is most commonly used by overlays.
+
+```
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+let () = N.f() // OK
+```
+
+## `@_fixed_layout`
```

Comment:
https://forums.swift.org/t/swift-org-blog-library-evolution-in-swift/33785/14

Reply:
I have a fuzzy understanding, but I'd rather someone else with a clearer understanding describe it. It seems like this is used for both classes and protocols. cc @slavapestov ?

Reply:
Updated based on discussion with Slava.